File: SqlLite/app/src/main/java/com/example/sqllite/RecycleAdapter.java
package com.example.sqllite;


import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class RecycleAdapter extends RecyclerView.Adapter<RecycleAdapter.ViewHolder> {

    private Context context;
    private ArrayList<Cars> carsArrayList;
    Intent intent;


    public RecycleAdapter(Context context, ArrayList<Cars> carsArrayList){
        this.context = context;
        this.carsArrayList = carsArrayList;
    }

    @NonNull
    @Override
    public RecycleAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_card, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecycleAdapter.ViewHolder holder, int position) {
        Cars cars = carsArrayList.get(position);
        holder.carsModel.setText(cars.getModel_Car());
        holder.carsOpisanie.setText(cars.getOpisanie_Car());
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(context, SecondActivity.class);

                // below we are passing all our values.
                i.putExtra("ID", cars.getID_Car());
                i.putExtra("model", cars.getModel_Car());
                i.putExtra("opisanie", cars.getOpisanie_Car());

                // starting our activity.
                context.startActivity(i);
            }
        });
    }

    @Override
    public int getItemCount() {
        return carsArrayList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder{

        TextView carsModel, carsOpisanie;
        Button delete;

        ViewHolder(View view) {
            super(view);
            carsModel = view.findViewById(R.id.c_model);
            carsOpisanie = view.findViewById(R.id.c_opisanie);
            delete = view.findViewById(R.id.delete_car);
        }

    }
}

File: PaperDB/app/src/main/java/com/example/paperdb/RecycleAdapter.kt
package com.example.paperdb

import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.core.content.ContextCompat.startActivity
import androidx.recyclerview.widget.RecyclerView


class RecycleAdapter(val context: Context, val listCars : MutableList<Car>, val onClick : (Car?) -> Unit)
    : RecyclerView.Adapter<RecycleAdapter.ViewHolder>() {
    class ViewHolder(view : View) : RecyclerView.ViewHolder(view){
        val CarTextView_name = view.findViewById<TextView>(R.id.cars_name_textView);
        val CarTextView_model = view.findViewById<TextView>(R.id.cars_model_textView);
        fun bind(car: Car?, onClick: (Car?) -> Unit) {
            itemView.setOnClickListener {
                onClick(car)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_card, parent, false)

        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val carItem = listCars[position]

        holder.CarTextView_name.setText(carItem.name);
        holder.CarTextView_model.setText(carItem.model);


        holder.bind(carItem, onClick)
    }

    override fun getItemCount(): Int {
        return listCars.size
    }


}